sematic/ui/src/components/RunTabs.tsx

```diff
@@ -47,9 +46,16 @@ export default function RunTabs(props: {
     <Tab label="Grafana" value="grafana" />
   ) : null;
 
+  useEffect(()=>{
```

Comment:
It is better not to have this component to manage the behavior of `<ArtifactLists />`. It can be handled directly in the `<ArtifactLists />` itself. 

Secondly, let's avoid using direct DOM manipulation in a React app. I think the React way of doing this is to use `useRef`. We can either use a list of references managed by `<ArtifactLists />` and scroll to the selected one. ([A example here](https://stackoverflow.com/questions/54633690/how-can-i-use-multiple-refs-for-an-array-of-elements-with-hooks)) , or if we pursue less tricky, we can have a `ref` per each artifact item and each item inspect whether itself is being selected ,if yes, scroll itself into view.

Reply:
Okay I will look into moving the scrolling functions into `ArtifactLists`. One issue I had was that I needed to wait for the element to be rendered to initiate the scroll and I was unsure about doing it from inside `ArtifactList`. I will try to implement this using `useRef`, I just felt that DOM manipulation was easier.

Reply:
Sure, let's give it a try and see how hard it is. Let me know if you run into issues, and we can even punt it.